appnav/src/main/kotlin/io/element/android/appnav/di/MatrixSessionCache.kt

```diff
@@ -25,45 +26,61 @@ import javax.inject.Inject
 
 private const val SAVE_INSTANCE_KEY = "io.element.android.x.di.MatrixClientsHolder.SaveInstanceKey"
 
+/**
+ * In-memory cache for logged in Matrix sessions.
+ *
+ * This component contains both the [MatrixClient] and the [SyncOrchestrator] for each session.
+ */
 @SingleIn(AppScope::class)
-@ContributesBinding(AppScope::class)
-class MatrixClientsHolder @Inject constructor(
+@ContributesBinding(AppScope::class, boundType = MatrixClientProvider::class)
```

Comment:
Nit: the `boundType` can be removed, this class now implement only 1 interface :)